### ISSUE TITLE
TST: RC2 failures (particularly wcsaxes)

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#7130]
Testing RC2, I found:

```
5 failed, 10347 passed, 82 skipped, 57 xfailed, 2 xpassed, 2382 warnings in 217.71 seconds
```

Two of them, I have seen in RC1: #6309 (`test_format_coord_regression`) and #7057 (`test_of_address`)

However, the three below are new and they are all from `wcsaxes`:
```
_____________________ test_label_visibility_rules_default ______________________

ax = <astropy.visualization.wcsaxes.core.WCSAxes object at 0x7fa07d24c2e8>

    def test_label_visibility_rules_default(ax):
>       assert_label_draw(ax, True, True)

.../astropy/visualization/wcsaxes/tests/test_coordinate_helpers.py:44:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

ax = <astropy.visualization.wcsaxes.core.WCSAxes object at 0x7fa07d24c2e8>
x_label = True, y_label = True

    def assert_label_draw(ax, x_label, y_label):
        ax.coords[0].set_axislabel("Label 1")
        ax.coords[1].set_axislabel("Label 2")

        with patch.object(ax.coords[0].axislabels, 'set_position') as pos1:
            with patch.object(ax.coords[1].axislabels, 'set_position') as pos2:
                ax.figure.canvas.draw()

>       assert pos1.call_count == x_label
E       AssertionError: assert 5 == True
E        +  where 5 = <MagicMock name='set_position' id='140327270711192'>.call_count

.../astropy/visualization/wcsaxes/tests/test_coordinate_helpers.py:39: AssertionError
```

```
______________________ test_label_visibility_rules_ticks _______________________

ax = <astropy.visualization.wcsaxes.core.WCSAxes object at 0x7fa07d1d2eb8>

    def test_label_visibility_rules_ticks(ax):

        ax.coords[0].set_axislabel_visibility_rule('ticks')
        ax.coords[1].set_axislabel_visibility_rule('ticks')

        ax.coords[0].set_ticklabel_visible(False)
        ax.coords[1].set_ticks(values=[-9999]*u.deg)

>       assert_label_draw(ax, True, False)

.../astropy/visualization/wcsaxes/tests/test_coordinate_helpers.py:63:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

ax = <astropy.visualization.wcsaxes.core.WCSAxes object at 0x7fa07d1d2eb8>
x_label = True, y_label = False

    def assert_label_draw(ax, x_label, y_label):
        ax.coords[0].set_axislabel("Label 1")
        ax.coords[1].set_axislabel("Label 2")

        with patch.object(ax.coords[0].axislabels, 'set_position') as pos1:
            with patch.object(ax.coords[1].axislabels, 'set_position') as pos2:
                ax.figure.canvas.draw()

>       assert pos1.call_count == x_label
E       AssertionError: assert 5 == True
E        +  where 5 = <MagicMock name='set_position' id='140327270185056'>.call_count

.../astropy/visualization/wcsaxes/tests/test_coordinate_helpers.py:39: AssertionError
```

```
______________________ test_label_visibility_rules_always ______________________

ax = <astropy.visualization.wcsaxes.core.WCSAxes object at 0x7fa07d188400>

    def test_label_visibility_rules_always(ax):

        ax.coords[0].set_axislabel_visibility_rule('always')
        ax.coords[1].set_axislabel_visibility_rule('always')

        ax.coords[0].set_ticklabel_visible(False)
        ax.coords[1].set_ticks(values=[-9999]*u.deg)

>       assert_label_draw(ax, True, True)

.../astropy/visualization/wcsaxes/tests/test_coordinate_helpers.py:74:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

ax = <astropy.visualization.wcsaxes.core.WCSAxes object at 0x7fa07d188400>
x_label = True, y_label = True

    def assert_label_draw(ax, x_label, y_label):
        ax.coords[0].set_axislabel("Label 1")
        ax.coords[1].set_axislabel("Label 2")

        with patch.object(ax.coords[0].axislabels, 'set_position') as pos1:
            with patch.object(ax.coords[1].axislabels, 'set_position') as pos2:
                ax.figure.canvas.draw()

>       assert pos1.call_count == x_label
E       AssertionError: assert 5 == True
E        +  where 5 = <MagicMock name='set_position' id='140327269938344'>.call_count

.../astropy/visualization/wcsaxes/tests/test_coordinate_helpers.py:39: AssertionError
```

Full log: 
[zlog_astropy_rc2.txt](https://github.com/astropy/astropy/files/1674964/zlog_astropy_rc2.txt)

**Comment by pllim** [comment]
Note: RC1 testing on Windows did not reveal anything that Linux didn't, so I am not going to rerun this on Windows this time around for RC2 unless explicitly requested to do so.

**Comment by astrofrog** [comment]
Oh hey @Cadair look, it's the tests you wrote! :)

@pllim - can you post the header from pytest so I can see all the versions?

I can try and investigate shortly, but since it's not tripping our CI, we could always postpone a fix for 3.0.1

**Comment by pllim** [comment]
@astrofrog , full log is attached in the original post above (at the bottom).
